mpsl/include/mpsl_ecb.h

```diff
@@ -9,30 +9,47 @@
 
 #include <stdint.h>
 
-#define MPSL_ECB_KEY_LENGTH                (16)                       /**< ECB key length. */
-#define MPSL_ECB_CLEARTEXT_LENGTH          (16)                       /**< ECB cleartext length. */
+#define MPSL_ECB_KEY_LENGTH                (16)                        /**< ECB key length. */
+#define MPSL_ECB_CLEARTEXT_LENGTH          (16)                        /**< ECB cleartext length. */
 #define MPSL_ECB_CIPHERTEXT_LENGTH         (MPSL_ECB_CLEARTEXT_LENGTH) /**< ECB ciphertext length. */
 
-/**@brief AES ECB parameter typedefs */
-typedef uint32_t mpsl_ecb_key_t[MPSL_ECB_KEY_LENGTH/sizeof(uint32_t)]; /**< Encryption key type. */
-typedef uint8_t mpsl_ecb_cleartext_t[MPSL_ECB_CLEARTEXT_LENGTH];       /**< Cleartext data type. */
-typedef uint8_t mpsl_ecb_ciphertext_t[MPSL_ECB_CIPHERTEXT_LENGTH];     /**< Ciphertext data type. */
-
 /**@brief AES ECB data structure */
 typedef struct
 {
-  mpsl_ecb_key_t        key;            /**< Encryption key. */
-  mpsl_ecb_cleartext_t  cleartext;      /**< Cleartext data. */
-  mpsl_ecb_ciphertext_t ciphertext;     /**< Ciphertext data. */
+  uint32_t key[MPSL_ECB_KEY_LENGTH/sizeof(uint32_t)]; /**< Encryption key. */
+  uint8_t  cleartext[MPSL_ECB_CLEARTEXT_LENGTH];      /**< Cleartext data. */
+  uint8_t  ciphertext[MPSL_ECB_CIPHERTEXT_LENGTH];    /**< Ciphertext data. */
 } mpsl_ecb_hal_data_t;
 
+#define MPSL_ECB_NO_FLAGS  (0)
+#define MPSL_ECB_INPUT_LE  (1u << 0)
+#define MPSL_ECB_OUTPUT_LE (1u << 1)
+
+/**@brief Encrypts a block according to the specified parameters.
+ *
+ * @note This function will only return once the ciphertext has been generated.
+ * @note This function may only be used on nRF52 and nRF53 series devices.
+ * @note This function is deprecated. Do not use.
+ *
+ * @param p_ecb_data Pointer to ECB data structure. Its elements are assumed to be big-endian.
+ */
+void mpsl_ecb_block_encrypt(mpsl_ecb_hal_data_t *p_ecb_data);
+
 /**@brief Encrypts a block according to the specified parameters.
  *
  * @note This function will only return once the ciphertext has been generated.
+ * @note The plaintext must be in RAM.
  *
- * @param[in, out] p_ecb_data pointer to the ECB paramters' struct (two input
- *                            parameters and one output parameter).
+ * @param[in]  key        Encryption key
+ * @param[in]  cleartext  Plaintext to be encrypted
+ * @param[out] ciphertext Encrypted text
+ * @param[in]  flags      Any combination of the following flags:
+ *  - MPSL_ECB_INPUT_LE     They key and cleartext are in little-endian format.
+ *  - MPSL_ECB_OUTPUT_LE    The ciphertext will be returned in little-endian format.
  */
-void mpsl_ecb_block_encrypt(mpsl_ecb_hal_data_t * p_ecb_data);
+void mpsl_ecb_block_encrypt_extended(const uint8_t key[MPSL_ECB_KEY_LENGTH],
```

Comment:
We should have an entry in the changelog describing that we have added a new API. This can be done in a separate PR